ale/drivers/mex_drivers.py

```diff
@@ -323,75 +326,145 @@ def line_scan_rate(self):
-            # print(self.ephemeris_stop_time, self.center_ephemeris_time)
-
-            self._line_scan_rate = (lines, times, durations)
-        return self._line_scan_rate
+        if not hasattr(self, '_binary_lines'):
+            self.read_image_data()
+        return self._binary_lines
+
+
+    def read_image_data(self):
+        """
+        Reads data off of image and stores in binary_exposure_durations, binary_lines,
+        and binary_ephemeris_times.
+
+        For HRSC, the exposure durations and ephemeris times are imbedded in the binary
+        data of the image itself. Each line is stored in what is referred to as a
+        "record" within the image. The label will have the size of each record,
+        the number of records, and the number of records in the label, so the
+        beginning of binary data can be calculated.
+
+        For each line/record of the binary data, the first 8 bytes make up the
+        double presicion value of the ephemeris time, with the next 4 bytes
+        making up the float value of the line exposure duration for the
+        associated line. NOTE: The prefix data is always LSB, regardless
+        of the overall file format.
+        """
+        lines = []
+        times = []
+        durations = []
+
+        with open(self._file, 'rb') as image_file:
+            bytes_per_record = self.label['RECORD_BYTES']
+            num_records = self.label['FILE_RECORDS']
+            img_start_record = self.label['^IMAGE']
+            img_start_byte = bytes_per_record * (img_start_record - 1) # Offset by one for zero-based records
+            num_img_records = num_records - img_start_record
+            image_file.seek(img_start_byte)
+
+            for record in range(num_img_records):
+                record_bytes = image_file.read(bytes_per_record)
+                eph_time = struct.unpack('<d', record_bytes[:8])[0]
+                exp_dur = struct.unpack('<f', record_bytes[8:12])[0] / 1000
+                # Offset for zero-based corrections, and then offest for ISIS pixel definition
+                lines.append(record+1-0.5)
+                times.append(eph_time)
+                durations.append(exp_dur)
+
+        self._binary_exposure_durations = durations
+        self._binary_lines = lines
+        self._binary_ephemeris_times = times
+
+
+
     @property
     def center_ephemeris_time(self):
-        if not hasattr(self, '_center_ephemeris_time'):
-            self._center_ephemeris_time = (self.ephemeris_stop_time - self._ephemeris_start_time) / 2
-        return self._center_ephemeris_time
+        """
+        Returns the center ephemeris time.
+
+        For HRSC, the center ephemeris time is calculated with the ephemeris stop time,
+        which is calculated from the binary image data.
+
+        Returns
+        -------
+        : float
+          Center ephemeris time
+        """
+        return (self.ephemeris_stop_time + self.ephemeris_start_time) / 2
+
```

Comment:
This is already implemented in base https://github.com/USGS-Astrogeology/ale/blob/master/ale/base/base.py#L308